src/webgpu/gpu_test.ts

```diff
@@ -517,13 +529,23 @@ export class GPUTest extends Fixture<GPUTestSubcaseBatchState> {
       layout?: TextureLayoutOptions;
     }
   ): void {
+    assert(
+      slice === 0 || dimension === '2d',
+      'texture slices are only implemented for 2d textures'
+    );
+
     format = resolvePerAspectFormat(format, layout?.aspect);
     const { byteLength, minBytesPerRow, bytesPerRow, rowsPerImage, mipSize } = getTextureCopyLayout(
       format,
       dimension,
       size,
       layout
     );
+    const copySize = [
+      mipSize[0],
+      dimension !== '1d' ? mipSize[1] : 1,
```

Comment:
nit: can we just assert (or assume) that if dimension is 1d, then mipSize[1] is 1? I would think that's true...
Similar on the next line.

Reply:
For 1d we can, but for 2d we cannot because of 2d texture arrays. The underlying helper just blindly uses the value in the third dimension for 2d textures, and that causes the copy size to be wrong when we are dealing with a slice of the 2d texture array. I added a comment and a TODO to fix the underlying helper in another change.